src/main/java/com/gtnewhorizon/gtnhlib/client/renderer/postprocessing/CustomFramebuffer.java

```diff
@@ -0,0 +1,571 @@
+                            GL11.GL_TEXTURE_2D,
+                            this.depthAttachment,
+                            0);
+                }
+            } else {
+                this.depthAttachment = OpenGlHelper.func_153185_f();
+                OpenGlHelper.func_153176_h(GL30.GL_RENDERBUFFER, this.depthAttachment);
+                if (stencil) {
+                    OpenGlHelper.func_153186_a(
+                            GL30.GL_RENDERBUFFER,
+                            EXTPackedDepthStencil.GL_DEPTH24_STENCIL8_EXT,
+                            framebufferWidth,
+                            framebufferHeight);
+                    OpenGlHelper.func_153190_b(
+                            GL30.GL_FRAMEBUFFER,
+                            EXTFramebufferObject.GL_DEPTH_ATTACHMENT_EXT,
+                            GL30.GL_RENDERBUFFER,
+                            this.depthAttachment);
+                    OpenGlHelper.func_153190_b(
+                            GL30.GL_FRAMEBUFFER,
+                            EXTFramebufferObject.GL_STENCIL_ATTACHMENT_EXT,
+                            GL30.GL_RENDERBUFFER,
+                            this.depthAttachment);
+                } else {
+                    OpenGlHelper.func_153186_a(
+                            GL30.GL_FRAMEBUFFER,
+                            GL14.GL_DEPTH_COMPONENT24,
+                            framebufferWidth,
+                            framebufferHeight);
+                    OpenGlHelper.func_153190_b(
+                            GL30.GL_FRAMEBUFFER,
+                            GL30.GL_DEPTH_ATTACHMENT,
+                            GL30.GL_RENDERBUFFER,
+                            this.depthAttachment);
+                }
+            }
+        }
+
+        clearCurrentFramebuffer();
+    }
+
+    public int createFramebufferAttachment() {
+        return createFramebufferAttachment(0);
+    }
+
+    public int createFramebufferAttachment(int slot) {
+        int texture = GL11.glGenTextures();
+
+        GL11.glBindTexture(GL11.GL_TEXTURE_2D, texture);
+        GL11.glTexImage2D(
+                GL11.GL_TEXTURE_2D,
+                0,
+                getTextureFormat(),
+                this.framebufferWidth,
+                this.framebufferHeight,
+                0,
+                GL11.GL_RGBA,
+                GL11.GL_UNSIGNED_BYTE,
+                (ByteBuffer) null);
+        final int filter = getTextureFilter();
+        GL11.glTexParameterf(GL11.GL_TEXTURE_2D, GL11.GL_TEXTURE_MIN_FILTER, filter);
+        GL11.glTexParameterf(GL11.GL_TEXTURE_2D, GL11.GL_TEXTURE_MAG_FILTER, filter);
+        GL11.glTexParameterf(GL11.GL_TEXTURE_2D, GL11.GL_TEXTURE_WRAP_S, GL11.GL_CLAMP);
+        GL11.glTexParameterf(GL11.GL_TEXTURE_2D, GL11.GL_TEXTURE_WRAP_T, GL11.GL_CLAMP);
+
+        // glFramebufferTexture2D
+        OpenGlHelper
+                .func_153188_a(GL30.GL_FRAMEBUFFER, GL30.GL_COLOR_ATTACHMENT0 | slot, GL11.GL_TEXTURE_2D, texture, 0);
```

Comment:
Should this use addition instead of bitwise-or?

```suggestion
                .func_153188_a(GL30.GL_FRAMEBUFFER, GL30.GL_COLOR_ATTACHMENT0 + slot, GL11.GL_TEXTURE_2D, texture, 0);
```

Reply:
Seems reasonable. I'd add a comment for that because otherwise no one will know how it's meant to be used.